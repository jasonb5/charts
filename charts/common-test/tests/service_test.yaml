suite: test service
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should expose ports from extraContainers
    set:
      workload: deployment

      extraContainers:
        db:
          image:
            repository: mariadb
            tag: latest

          service:
            ports:
              mariadb:
                port: 3306
    asserts:
      - equal:
          path: spec.template.spec.containers[1].ports
          value:
            - containerPort: 3306
              name: mariadb
              protocol: TCP
        documentIndex: 0
      - equal:
          path: spec
          value:
            ports:
              - name: default
                port: null
                protocol: TCP
                targetPort: null
              - name: mariadb
                port: 3306
                protocol: TCP
                targetPort: 3306
            selector:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/name: common-test
              common/workload: default
            type: ClusterIP
        documentIndex: 1
      - hasDocuments:
          count: 2
  - it: should overwrite defaults
    set:
      workload: deployment

      service:
        loadBalancerIP: 192.168.0.1
        type: ClusterIP
        ports:
          default:
            nodePort: 80
            port: 80
            exposedPort: 8080
    asserts:
      - equal:
          path: spec.template.spec.containers[0].ports
          value:
            - containerPort: 80
              name: default
              protocol: TCP
        documentIndex: 0
      - equal:
          path: spec
          value:
            loadBalancerIP: 192.168.0.1
            ports:
              - name: default
                nodePort: 80
                port: 8080
                protocol: TCP
                targetPort: 80
            selector:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/name: common-test
              common/workload: default
            type: ClusterIP
        documentIndex: 1
      - hasDocuments:
          count: 2
  - it: should enable service
    set:
      workload: deployment

      service:
        ports:
          default:
            port: 80
    asserts:
      - equal:
          path: spec.template.spec.containers[0].ports
          value:
            - containerPort: 80
              name: default
              protocol: TCP
        documentIndex: 0
      - equal:
          path: spec
          value:
            ports:
              - name: default
                port: 80
                protocol: TCP
                targetPort: 80
            selector:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/name: common-test
              common/workload: default
            type: ClusterIP
        documentIndex: 1
      - hasDocuments:
          count: 2
  - it: should disable service without workload
    set:
      service:
        ports:
          default:
            port: 80
    asserts:
      - hasDocuments:
          count: 0
