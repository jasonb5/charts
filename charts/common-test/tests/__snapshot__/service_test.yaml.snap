should expose ports from extraContainers:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      ports:
        - name: default
          port: null
          protocol: TCP
          targetPort: null
        - name: mariadb
          port: 3306
          protocol: TCP
          targetPort: 3306
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
        common/workload: default
      type: ClusterIP
  2: |
    - image: busybox:0.1.0
      livenessProbe:
        tcpSocket:
          port: default
      name: common-test
      ports:
        - containerPort: null
          name: default
          protocol: TCP
      readinessProbe:
        tcpSocket:
          port: default
    - image: mariadb:latest
      name: db
      ports:
        - containerPort: 3306
          name: mariadb
          protocol: TCP
should overwrite defaults:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      loadBalancerIP: 192.168.0.1
      ports:
        - name: default
          nodePort: 80
          port: 8080
          protocol: TCP
          targetPort: 80
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
        common/workload: default
      type: ClusterIP
  2: |
    - containerPort: 80
      name: default
      protocol: TCP
should pass:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      ports:
        - name: default
          port: 80
          protocol: TCP
          targetPort: 80
        - name: https
          port: 443
          protocol: TCP
          targetPort: 443
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
        common/workload: default
      type: ClusterIP
  2: |
    - containerPort: 80
      name: default
      protocol: TCP
    - containerPort: 443
      name: https
      protocol: TCP
